contracts/pox-delegation.clar

```diff
@@ -65,48 +64,45 @@
+(define-private (merge-details (stacker {lock-amount: uint, stacker: principal, unlock-burn-height: uint}) (user {pox-addr: {hashbytes: (buff 32), version: (buff 1)}, cycle: uint}))
   {lock-amount: (get lock-amount stacker),
   stacker: (get stacker stacker),
   unlock-burn-height: (get unlock-burn-height stacker),
   pox-addr: (get pox-addr user),
-  cycle: (get cycle user),
-  lock-period: (get lock-period user)})
+  cycle: (get cycle user)})
 
-(define-private (insert-in-new-list (pool principal) (reward-cycle uint) (last-index uint) (details {lock-amount: uint, stacker: principal, unlock-burn-height: uint, pox-addr: (tuple (hashbytes (buff 32)) (version (buff 1))), cycle: uint, lock-period: uint}))
+(define-private (insert-in-new-list (pool principal) (reward-cycle uint) (last-index uint) (details {lock-amount: uint, stacker: principal, unlock-burn-height: uint, pox-addr: (tuple (hashbytes (buff 32)) (version (buff 1))), cycle: uint}))
   (let ((index (+ last-index u1)))
-    (map-insert grouped-stackers (print {pool: pool, reward-cycle: reward-cycle, lock-period: (get lock-period details), index: index}) (list details))
-    (map-set grouped-stackers-len {pool: pool, reward-cycle: reward-cycle, lock-period: (get lock-period details)} index)))
+    (map-insert grouped-stackers (print {pool: pool, reward-cycle: reward-cycle, index: index}) (list details))
+    (map-set grouped-stackers-len {pool: pool, reward-cycle: reward-cycle} index)))
 
 
-;; Get the number of lists of stackers that have locked their stx for the given pool, cycle and lock-period.
-(define-read-only (get-status-lists-last-index (pool principal) (reward-cycle uint) (lock-period uint))
-  (default-to u0 (map-get? grouped-stackers-len {pool: pool, reward-cycle: reward-cycle, lock-period: lock-period}))
+;; Get the number of lists of stackers that have locked their stx for the given pool and cycle.
+(define-read-only (get-status-lists-last-index (pool principal) (reward-cycle uint))
+  (default-to u0 (map-get? grouped-stackers-len {pool: pool, reward-cycle: reward-cycle}))
 )
 
-(define-private (map-set-details (pool principal) (details {lock-amount: uint, stacker: principal, unlock-burn-height: uint, pox-addr: (tuple (hashbytes (buff 32)) (version (buff 1))), cycle: uint, lock-period: uint}))
+(define-private (map-set-details (pool principal) (details {lock-amount: uint, stacker: principal, unlock-burn-height: uint, pox-addr: (tuple (hashbytes (buff 32)) (version (buff 1))), cycle: uint}))
   (let ((reward-cycle (+ (contract-call? 'SP000000000000000000002Q6VF78.pox-2 current-pox-reward-cycle) u1))
-        (lock-period (get lock-period details))
-        (last-index (get-status-lists-last-index pool reward-cycle lock-period))
-        (key {pool: pool, reward-cycle: reward-cycle, lock-period: lock-period, index: last-index}))
+        (last-index (get-status-lists-last-index pool reward-cycle))
+        (key {pool: pool, reward-cycle: reward-cycle, index: last-index}))
```

Comment:
Can this `key`variable be inlined in the next line? I don't see it being used anywhere else and I think it would be easier to read. Otherwise, it could have a more descriptive name such as `grouped-stackers-map-key` or something similar. What do you think?

Reply:
![image](https://user-images.githubusercontent.com/1449049/226207366-2fa9fb5b-d688-4188-b71a-2c029800ecec.png)

key is used at three locations.
Let's rename it.